MekHQ/src/mekhq/campaign/enums/PlanetaryAcquisitionFactionLimit.java

```diff
@@ -0,0 +1,111 @@
+ *
+ * This file is part of MekHQ.
+ *
+ * MekHQ is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * MekHQ is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with MekHQ. If not, see <http://www.gnu.org/licenses/>.
+ */
+package mekhq.campaign.enums;
+
+import megamek.common.util.EncodeControl;
+import mekhq.MekHQ;
+
+import java.util.ResourceBundle;
+
+public enum PlanetaryAcquisitionFactionLimit {
+    //region Enum Declarations
+    ALL("PlanetaryAcquisitionFactionLimit.ALL.text"),
+    NEUTRAL("PlanetaryAcquisitionFactionLimit.NEUTRAL.text"),
+    ALLIED("PlanetaryAcquisitionFactionLimit.ALLIED.text"),
+    SELF("PlanetaryAcquisitionFactionLimit.SELF.text");
+    //endregion Enum Declarations
+
+    //region Variable Declarations
+    private final String name;
+
+    private final ResourceBundle resources = ResourceBundle.getBundle("mekhq.resources.Campaign", new EncodeControl());
+    //endregion Variable Declarations
+
+    //region Constructors
+    PlanetaryAcquisitionFactionLimit(final String name) {
+        this.name = resources.getString(name);
+    }
+    //endregion Constructors
+
+    //region Boolean Comparison Methods
+    public boolean isAll() {
+        return this == ALL;
+    }
+
+    public boolean isNeutral() {
+        return this == NEUTRAL;
+    }
+
+    public boolean isAllied() {
+        return this == ALLIED;
+    }
+
+    public boolean isSelf() {
+        return this == SELF;
+    }
+
+    public boolean generateOnEnemyPlanets() {
+        return isAll();
+    }
+
+    public boolean generateOnNeutralPlanets() {
+        return generateOnEnemyPlanets() || isNeutral();
+    }
+
+    public boolean generateOnAlliedPlanets() {
+        return generateOnNeutralPlanets() || isAllied();
+    }
+    //endregion Boolean Comparison Methods
+
+    //region File I/O
+    public static PlanetaryAcquisitionFactionLimit parseFromString(final String text) {
+        try {
+            return valueOf(text);
+        } catch (Exception ignored) {
+
+        }
+
+        try {
+            switch (Integer.parseInt(text)) {
+                case 0:
+                    return ALL;
+                case 2:
+                    return ALLIED;
+                case 3:
+                    return SELF;
+                case 1:
+                default:
+                    return NEUTRAL;
+            }
+        } catch (Exception ignored) {
+
+        }
+
+        MekHQ.getLogger().error("Unable to parse a PlanetaryAcquisitionFactionLimit from " + text
```

Comment:
Why not put this and the return into the catch? An empty catch looks weird.

Reply:
It's been the style I've been using for all of these, mostly just to keep the first and second trys looking the same. It would look odd if this enum was a different style to the rest of them.

Reply:
On that subject, why do we have valueOf followed up by a parseInt? Under what conditions would valueOf fail to return a meaningful value but a parseInt succeed?

Reply:
Legacy parsing. We currently save it as an integer. This replaces it with an enum, which saves using the name.

Reply:
How about enclosing both parse attempts in the same try/catch block, since it doesn't really add anything to have two separate try/catch blocks?

Reply:
Actually, the two are required. ValueOf exceptions out if it isn't valid, which then goes to the integer try catch